File: backend/src/main/java/com/examreg/examreg/models/Exam.java
package com.examreg.examreg.models;

import java.time.LocalDate;
import java.util.List;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import lombok.Data;

@Data
@Entity
public class Exam {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String examName;

  private LocalDate startDate;

  private LocalDate endDate;

  private String examStatus;

  @Column(columnDefinition = "TEXT")
  private String description; 

  @OneToMany(mappedBy = "exam")
  private List<Subject> subjects;
}

File: frontend/src/models/Exam.js
export class ExamResponse {
  constructor({ id, examName, startDate, endDate, examStatus, description }) {
    this.id = id
    this.examName = examName
    this.startDate = startDate
    this.endDate = endDate
    this.examStatus = examStatus
    this.description = description
  }

  static fromJSON(json) {
    return new ExamResponse(json)
  }
}
